examples/allennlp/allennlp_simple.py

```diff
@@ -100,26 +107,35 @@ def objective(trial):
     lr = trial.suggest_float("lr", 1e-1, 1e0, log=True)
     optimizer = torch.optim.SGD(model.parameters(), lr=lr)
 
-    iterator = allennlp.data.iterators.BasicIterator(batch_size=10,)
-    iterator.index_with(vocab)
+    data_loader = torch.utils.data.DataLoader(
+        train_dataset, batch_size=10, collate_fn=allennlp.data.allennlp_collate
+    )
+    validation_data_loader = torch.utils.data.DataLoader(
+        valid_dataset, batch_size=10, collate_fn=allennlp.data.allennlp_collate
+    )
 
     serialization_dir = os.path.join(MODEL_DIR, "trial_{}".format(trial.number))
-    trainer = allennlp.training.Trainer(
+    trainer = allennlp.training.GradientDescentTrainer(
         model=model,
         optimizer=optimizer,
-        iterator=iterator,
-        train_dataset=train_dataset,
-        validation_dataset=valid_dataset,
-        patience=3,
-        num_epochs=6,
+        data_loader=data_loader,
+        validation_data_loader=validation_data_loader,
+        validation_metric="+" + TARGET_METRIC,
+        patience=None,  # `patience=None` since it could conflict with AllenNLPPruningCallback
+        num_epochs=50,
         cuda_device=DEVICE,
         serialization_dir=serialization_dir,
+        epoch_callbacks=[AllenNLPPruningCallback(trial, "validation_" + TARGET_METRIC)],
     )
     metrics = trainer.train()
-    return metrics["best_validation_accuracy"]
+    return metrics["best_validation_" + TARGET_METRIC]
 
 
 if __name__ == "__main__":
+    random.seed(41)
+    torch.manual_seed(41)
+    numpy.random.seed(41)
+
     study = optuna.create_study(direction="maximize")
```

Comment:
I encountered nan-value errors during the training of AllenNLP model. This seems to happen stochastically, so how about fixing the seed value of `TPESampler`?

```
File "examples/allennlp/allennlp_simple.py", line 130, in objective
metrics = trainer.train()
File "/Users/yanase/.pyenv/versions/optuna-dev/lib/python3.7/site-packages/allennlp/training/trainer.py", line 802, in train
train_metrics = self._train_epoch(epoch)
File "/Users/yanase/.pyenv/versions/optuna-dev/lib/python3.7/site-packages/allennlp/training/trainer.py", line 569, in _train_epoch
raise ValueError("nan loss encountered")
ValueError: nan loss encountered
```

```suggestion
    study = optuna.create_study(direction="maximize", sampler=optuna.samplers.TPESampler(seed=41))
```